pkg/internal/predicates/predicates.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package predicates
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	crtpredicate "sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var log = logf.Log.WithName("predicates")
+
+// PredicateFuncs returns functions defined for filtering events
+func PredicateFuncs() crtpredicate.Funcs {
```

Comment:
Nit: Can we rename this to `DependentPredicateFuncs()` in case we need to add other predicates to this package?

LGTM once that is addressed and CI passes.

Reply:
@joelanford , will update the code as suggested. Thanks